apis/core/v1beta1/featureflag_webhook.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"encoding/json"
+	"fmt"
+	"sync"
+
+	_ "embed"
+
+	schema "github.com/open-feature/flagd-schemas/json"
+	"github.com/xeipuuv/gojsonschema"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+// log is for logging in this package.
+var featureFlagLog = logf.Log.WithName("featureflag-resource")
+var compiledSchema *gojsonschema.Schema
+var schemaInitOnce sync.Once
+
+func (ff *FeatureFlag) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(ff).
+		Complete()
+}
+
+//+kubebuilder:webhook:path=/validate-core-openfeature-dev-v1beta1-featureflag,mutating=false,failurePolicy=fail,sideEffects=None,groups=core.openfeature.dev,resources=featureflags,verbs=create;update,versions=v1beta1,name=vfeatureflag.kb.io,admissionReviewVersions=v1
+
+var _ webhook.Validator = &FeatureFlag{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (ff *FeatureFlag) ValidateCreate() error {
+	featureFlagLog.Info("validate create", "name", ff.Name)
```

Comment:
Is it standard to have these as `Info` level? If it is, that's fine, I'm just curious. I seems like more of a `Debug` thing... but maybe this is standard for better logging of resource changes? I suspect you know better than me.

Reply:
This part was auto generated by kubebuilder. I don't have a strong opinion for this, but it can be changed to debug mode if you like

Reply:
Use your judgement, I just wanted to point it out.